src/pages/instances/actions/InstanceBulkAction.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FC, ReactNode } from "react";
+import ConfirmationButton from "components/ConfirmationButton";
+import {
+  LxdInstance,
+  LxdInstanceAction,
+  LxdInstanceStatus,
+} from "types/instance";
+import {
+  instanceAction,
+  instanceActionLabel,
+  instanceActions,
+  pluralizeInstance,
+  statusLabel,
+} from "util/instanceBulkActions";
+import { ICONS, ValueOf } from "@canonical/react-components";
+
+interface Props {
+  action: LxdInstanceAction;
+  icon?: ValueOf<typeof ICONS> | string;
+  instances: LxdInstance[];
+  isLoading: boolean;
+  isDisabled: boolean;
+  confirmationLabel: string;
+  onClick: () => void;
+  confirmationAppearance?: string;
+  confirmationExtra?: ReactNode;
+}
+
+const InstanceBulkAction: FC<Props> = ({
+  action,
+  confirmationAppearance,
+  confirmationExtra,
+  confirmationLabel,
+  icon,
+  instances,
+  isLoading,
+  isDisabled,
+  onClick,
+}) => {
+  const statusLine = (
+    currentState: LxdInstanceStatus,
+    desiredAction: LxdInstanceAction
+  ) => {
+    const count = instances.filter(
+      (instance) => instance.status === currentState
+    ).length;
+
+    const status = statusLabel(currentState) ?? "";
+    const actionRaw = instanceAction(desiredAction, currentState);
+    const action = instanceActionLabel(actionRaw);
```

Comment:
```suggestion
    const action = actionRaw ? instanceActionLabel(actionRaw) : "ignored";
```
Which also has the benefit of making the code clearer in this context. So that when a few lines below we have `if (action === "ignored")`, we clearly know where that "ignored" is coming from, without needing to switch to the context of the util function.

Reply:
Good idea, though with the not allowing undefined actions this becomes even simpler!